### PR TITLE
AC 1.6 Schema Update for Dynamic Typeahead and Inline-Edit Features

### PR DESCRIPTION
**Pull request by SuryanshPrabhatMSFT**
# Summary of Changes
1. Updated the schema for v1.6 with the properties required in [Inline-edit of Input elements](https://github.com/microsoft/AdaptiveCards/pull/8143) and [Dynamic Typeahead in Input.Choiceset](https://github.com/microsoft/AdaptiveCards/pull/8267) features.
2. Added `count` and `skip` propery in `choices.data` as it can be sent in the card payload as well. These properties are sent to the bot as part of the invoke payload. Both of them do not affect anything else.
3. In the conditional block to determine whether to start the request to fetch dynamic options from the bot, we extract the text directly from the component itself and not from the parent. We do this because in the former way, it returned choice.value and not choice.title, if the text written by the user matched with one of the static choices.

## Schema Explorer Site
image
image
image
image

 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/8495)

### CODE REVIEW

**Comment by SuryanshPrabhatMSFT** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: schemas/1.6.0/adaptive-card.json [hunk 1]
```diff
@@ -1071,6 +1071,34 @@
         }
       ]
     },
+    "Data.Query": {
+      "description": "Data to be sent to the bot to fetch dynamic choices that can be displayed in the choiceset dropdown for selection",
```
Comment (anna-dingler) [inline schemas/1.6.0/adaptive-card.json:1075]: What do you mean by "bot" here?

File: schemas/1.6.0/adaptive-card.json [hunk 2]
```diff
@@ -2168,6 +2218,34 @@
...
+            "above"
+          ]
+        },
+        {
+          "pattern": "^([i|I][n|N][l|L][i|I][n|N][e|E])|([a|A][b|B][o|O][v|V][e|E])$"
+        }
+      ]
+    },
+    "InputStyle": {
+      "description": "`revealOnHover`: Show input fields in read only view unless user takes any action. `default`: Show the existing view for input controls",
```
Comment (anna-dingler) [inline schemas/1.6.0/adaptive-card.json:2236]: "existing view" could be a little clearer here. Maybe "default editable state"?

File: schemas/1.6.0/adaptive-card.json [hunk 3]
```diff
@@ -3279,6 +3357,28 @@
...
+        "labelWidth": {
+          "anyOf": [
+            {
+              "type": "string"
+            },
+            {
+              "type": "number"
+            }
+          ],
+          "description": "Determines the width of the label in percent like 40 or a specific pixel width like '40px' when label is placed inline with the input . labelWidth would be ignored when the label is displayed above the input.",
```
Comment (anna-dingler) [inline schemas/1.6.0/adaptive-card.json:3374]: Extra space before the first `.`?

**Comment by anna-dingler** [comment]
Hi @SuryanshPrabhatMSFT, were you able to verify the schema docs on the website?

File: schemas/1.6.0/adaptive-card.json (hunk 1 above)
Comment (SuryanshPrabhatMSFT) [reply schemas/1.6.0/adaptive-card.json:1075]: Rewrote the statement for more clarity.

File: schemas/1.6.0/adaptive-card.json (hunk 2 above)
Comment (SuryanshPrabhatMSFT) [reply schemas/1.6.0/adaptive-card.json:2236]: Updated.

File: schemas/1.6.0/adaptive-card.json (hunk 3 above)
Comment (SuryanshPrabhatMSFT) [reply schemas/1.6.0/adaptive-card.json:3374]: Updated.

**Comment by SuryanshPrabhatMSFT** [comment]
> Hi @SuryanshPrabhatMSFT, were you able to verify the schema docs on the website?

@anna-dingler Thanks for pointing this out. Confirmed the changes on the website. Attached the screenshots in the PR description. Please let me know if anything is missing.

**Comment by anna-dingler** [comment]
@SuryanshPrabhatMSFT, 

I think this looks great - thank you for addressing my comments. Just waiting for consensus from @jwoo-msft.

Thanks!

File: schemas/1.6.0/adaptive-card.json [hunk 4]
```diff
@@ -2168,6 +2222,34 @@
...
+            "above"
+          ]
+        },
+        {
+          "pattern": "^([i|I][n|N][l|L][i|I][n|N][e|E])|([a|A][b|B][o|O][v|V][e|E])$"
+        }
+      ]
+    },
+    "InputStyle": {
+      "description": "`revealOnHover`: Show input fields in read only view unless user takes any action. `default`: Show the default editable state for input controls",
```
Comment (jwoo-msft) [inline schemas/1.6.0/adaptive-card.json:2240]: we need to let mobile users know that this feature is not available to their users.

File: schemas/1.6.0/adaptive-card.json [hunk 5]
```diff
@@ -3279,6 +3361,28 @@
           "description": "Label for this input",
           "version": "1.3"
         },
+        "labelPosition": {
+          "$ref": "#/definitions/InputLabelPosition",
+          "description": "Determines the position of the label. It can take 'inline' and 'above' values. By default, the label is placed 'above' when label position is not specified.",
```
Comment (jwoo-msft) [inline schemas/1.6.0/adaptive-card.json:3366]: is the label position implemented for mobile renderers?

**Review by jwoo-msft — CHANGES_REQUESTED** [review]

File: schemas/1.6.0/adaptive-card.json (hunk 4 above)
Comment (SuryanshPrabhat) [reply schemas/1.6.0/adaptive-card.json:2240]: Indicated!

File: schemas/1.6.0/adaptive-card.json (hunk 5 above)
Comment (SuryanshPrabhat) [reply schemas/1.6.0/adaptive-card.json]: Indicated that the support is only in the JS SDK.

**Review by paulcam206 — APPROVED** [review]

**Review by jwoo-msft — APPROVED** [review]

**Pull request merged**